src/samples/iio+network/iio-gyroscope-console-and-mqtt-publish.c

```diff
@@ -0,0 +1,542 @@
+
+    out.x = out.x - cal_data->bias_x;
+    out.y = out.y - cal_data->bias_y;
+    out.z = out.z - cal_data->bias_z;
+#endif
+
+#ifdef DENOISE_MEDIAN
+    denoise_median(filter_data, &out.x, &out.y, &out.z);
+#endif
+
+#ifdef GYRO_CALIBRATE
+    clamp_gyro_readings_to_zero(cal_data, &out.x, &out.y, &out.z);
+#endif
+
+#ifdef GYRO_CALIBRATE
+    snprintf(buffer, 127, "%f\t%f\t%f\t[rad/sec]\t(%s)", out.x, out.y, out.z,
+        (cal_data->calibrated ? "Calibrated" : "Not calibrated"));
+#else
+    snprintf(buffer, 127, "%f\t%f\t%f\t[rad/sec]\t(Calibration disabled)",
+        out.x, out.y, out.z);
+#endif
+    printf("%s\n", buffer);
+
+#ifdef MQTT
+    if (gyro_data->mqtt == NULL) {
+        SOL_WRN("mqtt is null");
+        return;
+    }
+    if (sol_mqtt_get_connection_status(gyro_data->mqtt) == SOL_MQTT_CONNECTED) {
+
+        payload = SOL_BUFFER_INIT_CONST(buffer, strlen(buffer));
+
+        mqtt_message = (struct sol_mqtt_message){
+            SOL_SET_API_VERSION(.api_version = SOL_MQTT_MESSAGE_API_VERSION, )
+            .topic = gyro_data->mqtt_topic,
+            .payload = &payload,
+            .qos = SOL_MQTT_QOS_EXACTLY_ONCE,
+            .retain = false,
+        };
+
+        if (sol_mqtt_publish(gyro_data->mqtt, &mqtt_message)) {
+            SOL_WRN("Unable to publish message");
+        }
+    }
+#endif
+    return;
+
+error:
+    SOL_WRN("Could not read channel buffer values");
+}
+
+int
+main(int argc, char *argv[])
+{
+    struct sol_iio_device *device = NULL;
+    struct sol_iio_config iio_config;
+    struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
+    struct iio_gyroscope_data gyro_data;
+    int device_id;
+
+    memset(&iio_config, 0, sizeof(struct sol_iio_config));
+    memset(&gyro_data, 0, sizeof(struct iio_gyroscope_data));
+#ifdef MQTT
+    struct sol_mqtt_config mqtt_config = {
+        SOL_SET_API_VERSION(.api_version = SOL_MQTT_CONFIG_API_VERSION, )
+        .clean_session = true,
+        .keep_alive = 60,
+        .handlers = {
+            SOL_SET_API_VERSION(.api_version = SOL_MQTT_HANDLERS_API_VERSION, )
+            .connect = on_connect,
+            .disconnect = on_disconnect,
+        },
+    };
+#endif
+    sol_init();
+
+#ifdef MQTT
+    if (argc < 11) {
+        SOL_ERR("\nUsage: %s <device name> <trigger name> <buffer size> " \
+            "<sampling frequency> <scale> <custom offset> <offset> " \
+            "<MQTT broker ip> <MQTT broker port> <MQTT topic>\n" \
+            "\t<buffer size>:\t\t0=default\n" \
+            "\t<sampling frequency>:\t-1=default\n" \
+            "\t<scale>:\t\t<-1=default\n" \
+            "\t<custom offset>:\ty or n\n" \
+            "\t<offset>:\t\tonly take effect if custom offset is \"y\"\n" \
+            "Press CTRL + C to quit\n" \
+            , argv[0]);
+#else
+    if (argc < 8) {
+        SOL_ERR("\nUsage: %s <device name> <trigger name> <buffer size> " \
```

Comment:
For this kind of message to user, prefer normal `fprintf(stderr,`, to avoid cluttering message with line or file information.